src/global/variables/clauses/IntegerLiteral.java

```diff
@@ -1,26 +1,67 @@
 package global.variables.clauses;
 
+import global.utils.RegexUtil;
 import global.variables.Clause;
 
 public class IntegerLiteral extends Clause {
     int value;
+    int upper;
+    int lower;
 
     public IntegerLiteral(int value) {
         super();
         this.value = value;
+        this.upper = value;
+        this.lower = value;
+
     }
 
     public IntegerLiteral(int value, String name) {
         super(name);
         this.value = value;
+        this.upper = value;
+        this.lower = value;
+    }
+
+    public IntegerLiteral(int lower, int upper) {
+        super();
+        this.value = 0;
+        this.lower = lower;
+        this.upper = upper;
+    }
+
+    public IntegerLiteral(int lower, int upper, String name) {
+        super(name);
+        this.value = 0;
+        this.lower = lower;
+        this.upper = upper;
+    }
+
+    public int getUpper() {
+        return upper;
+    }
+
+    public int getLower() {
+        return lower;
     }
 
     public int getValue() {
         return value;
     }
 
+    public String getValueOrRegexRange() {
+        if (getLower() == getUpper()) {
+            return String.valueOf(getValue());
+        } else if (getLower() > getUpper()) {
+            //TODO: I'm not sure if having a default like this is a good idea, or if I should put it into the constructor
```

Comment:
If lower is above upper, then it's configured incorrectly and shouldn't be allowed to reach this step

Reply:
So clauses have a `.validate()` method in the abstract class that default just returns `true`. This can be overridden in this class. For context, the intent of `.validate()` is validating that the output retrieved for this clause is okay, so you can override it to be like "if a lower and upper are defined, return true if the output-retrieved-value is within that, if not then just default return true".

So this ^ handles checking that the number is in the range. We should also have a new constructor here that takes an array of integers so they can set a discrete set of potential values.

We should also make a new method in `Clause.java` called `.validateParams()` that checks if parameters are okay. It should return true by default but in cases like this, we override it to check that upper > lower and such. We then loop through all clauses and ensure that `.validateParams()` is okay (do this @beforeAll tests and raise a custom `ClauseException` if a clause is invalidly configured.